openquake/hazardlib/sourceconverter.py

```diff
@@ -1340,16 +1347,41 @@ def convert_sliplist(self, node):
             lst = [node_to_dict(n)['slip'] for n in slip_list.nodes]
         return str(lst)
 
+    def convert_rake(self, node):
+        try:
+            return ~node.rake
+        except AttributeError:
+            return ''
+
+    def convert_geomprops(self, node):
+        lst = [node_to_dict(n) for n in node.nodes
+               if striptag(n.tag) not in EXCLUDE_FROM_GEOM_PROPS]
+        return str(lst)
+
+    def convert_groupname(self, node):
+        try:
+            return str(node['groupname'])
+        except KeyError:
+            return ''
+
+    def convert_tectonicregion(self, node):
+        try:
+            return str(node['tectonicRegion'])
+        except KeyError:
+            return ''
+
     def convert_areaSource(self, node):
         geom = node.areaGeometry
         coords = split_coords_2d(~geom.Polygon.exterior.LinearRing.posList)
-        coords += [coords[0]]  # close the polygon
+        if coords[0] != coords[-1]:
+            coords += [coords[0]]  # close the polygon
         # TODO: area_discretization = geom.attrib.get('discretization')
```

Comment:
Please check this TODO

Reply:
I'm going to investigate about this. However, this TODO was not introduced in this branch.

Reply:
I improved the management of geometry-related properties, collecting also attributes (including discretization). I made it general, so it will be possible to add other attributes and have them automatically converted without further changes in the code.